src/test/java/lambda/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,94 @@
+package lambda.part1.exercise;
+
+import org.junit.Test;
+
+import java.util.StringJoiner;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"FieldCanBeLocal", "ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @FunctionalInterface
+    private interface Multiplier<T> {
+        T multiply(T value, int multiplier);
+
+        default T twice(T t) {
+            return multiply(t, 2);
+        }
+    }
+
+    private void testIntegerMultiplier(Multiplier<Integer> multiplier) {
+        assertEquals(6, multiplier.multiply(3, 2).intValue());
+        assertEquals(0, multiplier.multiply(Integer.MIN_VALUE, 0).intValue());
+        assertEquals(-7, multiplier.multiply(7, -1).intValue());
+        assertEquals(10, multiplier.twice(5).intValue());
+        assertEquals(0, multiplier.twice(0).intValue());
+    }
+
+    @Test
+    public void implementsIntegerMultiplierUsingAnonymousClass() {
+        Multiplier<Integer> multiplier = new Multiplier<Integer>() {
+            @Override
+            public Integer multiply(Integer value, int multiplier) {
+                return value*multiplier;
+            }
+        };
+
+        testIntegerMultiplier(multiplier);
+    }
+
+    @Test
+    public void implementsMultiplierUsingStatementLambda() {
+        Multiplier<Integer> multiplier = (value1, value2) -> {return value1*value2;};
```

Comment:
It's better to autoformat your code (select code and press Ctrl+Alt+Shift+L) before commit